src/lib/modules/console/index.ts

```diff
@@ -0,0 +1,226 @@
+  private cmdHistoryFile: string;
+  private suggestions: Suggestions;
+
+  constructor(embark: any, options: any) {
+    this.embark = embark;
+    this.events = options.events;
+    this.plugins = options.plugins;
+    this.version = options.version;
+    this.logger = options.logger;
+    this.ipc = options.ipc;
+    this.config = options.config;
+    this.history = [];
+    this.cmdHistoryFile = options.cmdHistoryFile || fs.dappPath(".embark", "cmd_history");
+    this.loadHistory();
+
+    if (this.ipc.isServer()) {
+      this.ipc.on("console:executeCmd", this.executeCmd.bind(this));
+    }
+    this.events.setCommandHandler("console:executeCmd", this.executeCmd.bind(this));
+    this.events.setCommandHandler("console:history", (cb: any) => this.getHistory(this.cmdHistorySize(), cb));
+    this.registerEmbarkJs();
+    this.registerConsoleCommands();
+    this.registerApi();
+
+    this.suggestions = new Suggestions(embark, options);
+  }
+
+  private cmdHistorySize() {
+    return env.anchoredValue(env.CMD_HISTORY_SIZE);
+  }
+
+  private registerApi() {
+    const plugin = this.plugins.createPlugin("consoleApi", {});
+    plugin.registerAPICall("post", "/embark-api/command", (req: any, res: any) => {
+      this.executeCmd(req.body.command, (err: any, result: any) => {
+        if (err) {
+          return res.send({result: err.message || err});
+        }
+        if (typeof result === "string") {
+          return res.send({result});
+        }
+        res.send({result: stringify(result, utils.jsonFunctionReplacer, 2)});
+      });
+    });
+  }
+
+  private processEmbarkCmd(cmd: string) {
+    if (cmd === "help" || cmd === __("help") || cmd === "01189998819991197253") {
+      const helpText = [
+        __("Welcome to Embark") + " " + this.version,
+        "",
+        __("possible commands are:"),
+        "versions - " + __("display versions in use for libraries and tools like web3 and solc"),
+        "history - " + __("display console commands history"),
+        // TODO: only if the blockchain is actually active!
+        // will need to pass te current embark state here
+        "ipfs - " + __("instantiated js-ipfs object configured to the current environment (available if ipfs is enabled)"),
+        "swarm - " + __("instantiated swarm-api object configured to the current environment (available if swarm is enabled)"),
+        "web3 - " + __("instantiated web3.js object configured to the current environment"),
+        "EmbarkJS - " + __("EmbarkJS static functions for Storage, Messages, Names, etc."),
+        "quit - " + __("to immediatly exit (alias: exit)"),
+        "",
+        __("The web3 object and the interfaces for the deployed contracts and their methods are also available"),
+      ];
+      return helpText.join("\n");
+    } else if (["quit", "exit", "sair", "sortir", __("quit")].indexOf(cmd) >= 0) {
+      utils.exit();
+    }
+    return false;
+  }
+
+  private executeCmd(cmd: string, callback: any) {
```

Comment:
Wouldn't `callback` be a function? (Disclaimer: I know nothing of TS)

Reply:
Instead of `any` I would suggest `void`, that is more precise and I think callback don't return anything?